### DISCUSSION TITLE
How can i increase the accuracy of the model output?

### DISCUSSION THREAD

**Discussion by Dreamown** [neperfepx/neper#642] [Q&A] [answered]
I found that the model files generated using neper, e.g. .tess, .obj, .geo, only store coordinate values up to 12 significant digits. When I use a custom method to convert .tess to .stp and then import it into rhino, the accuracy of the model is affected, and storing with more than 14 valid digits will not cause any errors.

Below is the maximum number of digits displayed for coordinates in the .tess file. I would like to be more precise.
![image](https://user-images.githubusercontent.com/32787082/212479169-a5237126-bddc-4c81-86d2-1234b4041168.png)

**Comment by rquey** [comment, marked as answer]
There is no option dedicated to this, but what you can do is substitute all `%.12f` by `%.14f` in the source code:
```
sed -i "s/%\.12f/%.14f/g" `find . -name "*.[c,cpp]"`
```
then recompile.

**Reply by Dreamown** [reply]
Thank you very much. It works very well.

**Reply by rquey** [reply]
@Dreamown In the next version, there will be the `REAL_PRINT_FORMAT` configuration variable for this. This will do what you want:
```
cmake -DREAL_PRINT_FORMAT=\"%.14f\" ..
```

**Reply by Dreamown** [reply]
Thank you very much for your help. neper is by far the best microstructural reconstruction software I've ever used.

**Reply by Dreamown** [reply]
I developed a plugin for the grasshopper platform to pass the neper model to rhino, and it seems to work fine, but it still needs further maintenance, maybe I'll expose it later. Thanks again to neper for your help.
![image](https://user-images.githubusercontent.com/32787082/215789649-c1a495b1-b040-4713-9eba-9364b3e7d0ef.png)
![image](https://user-images.githubusercontent.com/32787082/215789704-f0ba46e4-5c30-4faa-802e-b5c6afae2910.png)

**Reply by rquey** [reply]
Looks great. If it become available, I'll add it [here](https://neper.info/others.html#id1).

**Reply by Dreamown** [reply]
This is a good idea, I need to test it further and I will let you know first when it is suitable for release.

**Comment by Dreamown** [comment]
@rquey  This is the first version of the plugin, and I hope it will extend neper to rhino to make it easier to use.
https://www.food4rhino.com/en/app/neper2rhino?lang=en

**Reply by rquey** [reply]
Thanks. I've added the link. I would suggest to write the description of your plugin in English so that everyone can read it.

> This is a micro plug-in to quickly convert Neper-generated mesh models into analytical models and to facilitate importing models before CPFEM. Neper is a very powerful microstructure segmentation software, and a detailed description can be found on the official website (https://neper.info/).
>
> Notes.
>
> 1. first, in neper you need to use the command "-format obj" to convert your tess file into an obj file.
>
> 2. Then, import the obj file into Rhino and use this plugin to convert the mesh into a parsed model.
>
> 3. Finally, export the parsed model and save it as a stp file. Or, you can further modify the analytical model in Grasshopper to get the microstructure you need.
>
> 4. Neper2Rhino is located under the "Mesh -> Analysis" category.
>
> 5. Please add the plugin in "C:\Users\XXX\AppData\Roaming\Grasshopper\UserObjects".

**Comment by jdavidgg01** [comment]
Hi @Dreamown , is it possible that you can expand your instructions or if you can upload a short video that would be great, I'm using Rhino/grasshoper for the first time and I don't know how to use your model, I tried many times but I think I'm doing something wrong because the step files is not being exported
